Added release notes for v5.3.6 (#13)
hazelcast/clc-docs: docs/modules/ROOT/pages/release-notes-5.3.6.adoc

= 5.3.6 Release Notes

== New Features

> === AtomicRef support

Reviewer: ```suggestion
=== AtomicRef Support
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

Added the following `atomicreference` commands:

* `clear`
* `compare-and-set`
* `contains`
* `destroy`
* `get`
* `get-and-set`
* `is-null`
* `set`

The following advanced script functions are also supported:

* `atomicreference_clear(name='default')`
* `atomicreference_compare_and_set(expected:any, value:any, name='default')`
* `atomicreference_contains(value:any, name='default')`
* `atomicreference_destroy(name='default')`
* `atomicreference_get(name='default')`
* `atomicreference_get_and_set(value:any, name='default')`
* `atomicreference_is_null(name='default')`
* `atomicreference_set(value:any, name='default')`

> === Viridian advanced script functions

Reviewer: ```suggestion
=== Viridian Advanced Script Functions
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* `create_cluster(name:string, development=False, prerelease=False, wait=False) -> Cluster`
* `list_clusters() -> list(Cluster)`
* `delete_cluster(cluster:id | name | Cluster)`
* `get_cluster(cluster:id | name | Cluster) -> Cluster`
* `import_config(cluster:id | name | Cluster, name='') -> path`
* `stop_cluster(cluster:id | name | Cluster, wait=False)`
* `resume_cluster(cluster:id | name | Cluster, wait=False)`
* `list_custom_classes(cluster:id | name | Cluster) -> list(Class)`
* `upload_custom_class(cluster:id | name | Cluster, path:string, wait=False)`
* `download_custom_class(cluster:id | name | Cluster, artifact:any, output_path:string) -> path`
* `delete_custom_class(cluster:id | name | Cluster, artifact_id:any, wait=False)`
* `download_logs(cluster:id | name | Cluster, dir:string) -> list(path)`

Added the `viridian_cluster` type with the following attributes:

* `id`
* `name`
* `cluster_type`
* `state`
* `desired_state`
* `creation_time`
* `start_time`
* `client_count`
* `hot_backup_enabled`
* `hot_restart_enabled`
* `ip_white_list_enabled`

Add `viridian_custom_class` type with the following attributes:

* `id`
* `name`
* `status`
* `temporary_id`
* `generated_filename`

> === Job advanced script functions

Reviewer: ```suggestion
=== Job Advanced Script Functions
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* `job_submit(path:string, params:string*, name='', snapshot='', class_name='')`
* `job_list(include_sql=False, include_user_cancelled=False) -> list(dict)`
* `job_cancel(job_name_or_id:string, force=False)`
* `job_restart(job_name_or_id:string, force=False)`
* `job_suspend(job_name_or_id:string, force=False)`
* `job_resume(job_name_or_id:string)`
* `job_export_snapshot(job_name_or_id:string, name:string, cancel=False)`

> === Snapshot advanced script functions

Reviewer: ```suggestion
=== Snapshot Advanced Script Functions
```

* `snapshot_list() -> list(dict)`
* `snapshot_delete(name:string)`

> === Configuration advanced script functions

Reviewer: ```suggestion
=== Configuration Advanced Script Functions
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* `config_add(config_name_or_path:string, config:dict) -> path`
* `config_list() -> list(config_name)`

> === List advanced script functions

Reviewer: ```suggestion
=== List Advanced Script Functions
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* `list_add(value:any, name='default', index=0)`
* `list_clear(name='default')`
* `list_contains(value:any, name='default')`
* `list_destroy(name='default')`
* `list_remove_index(index:int, name='default')`
* `list_remove_value(value:any, name='default')`
* `list_set(index:int, value:any, name='default')`
* `list_size(name='default')`

> === MultiMap advanced script functions

Reviewer: ```suggestion
=== MultiMap Advanced Script Functions
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* `multimap_clear(name='default')`
* `multimap_destroy(name='default')`
* `multimap_entry_set(name='default')`
* `multimap_get(key:any, name='default')`
* `multimap_key_set(name='default')`
* `multimap_lock(key:any, name='default', ttl=-1)`
* `multimap_put(key:any, value:any, name='default')`
* `multimap_remove(key:any, name='default')`
* `multimap_size(name='default')`
* `multimap_try_lock(key:any, name='default', ttl=-1)`
* `multimap_unlock(key:any, name='default')`
* `multimap_values(name='default')`

> === Queue advanced script functions

Reviewer: ```suggestion
=== Queue Advanced Script Functions
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* `queue_clear(name='default')`
* `queue_destroy(name='default')`
* `queue_offer(value:any*, name='default')`
* `queue_poll(name='default')`
* `queue_size(name='default')`

> === Topic advanced script functions

Reviewer: ```suggestion
=== Topic Advanced Script Functions
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* `topic_destroy(name='default')`
* `topic_publish(value:any+, name='default')`

> === Other advanvced script functions

Reviewer: ```suggestion
=== Other Advanvced Script Functions
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* `object_list(dds:string?, show_hidden=False) -> list(object_name) | list(service_name, object_name)`
* `exit(code:int)`
* `home() -> path`
* `version() -> string`
* `write(value:any, location:string, format='')`
* `read(location:string, format='default')`
* `write_data(data:data, location:string)`
* `read_data(location:string)`
* `decode_data(value:data)`

> === Other features

Reviewer: ```suggestion
=== Other Features
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* Added doc strings support to advanced scripts. All lines starting with `# doc: ` are doc strings.
* Added `script list` command.
* Added the `sql` advanced script command.

> * Added the `file` template action which can read from a file with one of the extensions: `.yaml`, `.json` or `.txt`.

Reviewer: ```suggestion
* Added the `file` template action, which can read from a file with an extension of `.yaml`, `.json` or `.txt`.
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

> * Added the `\exit` and `\echo` commands. These commands are available in the script and interactive modes. See: [echo](https://docs.hazelcast.com/clc/latest/echo) and [exit](https://docs.hazelcast.com/clc/latest/exit) command documentation.

Reviewer: ```suggestion
* Added the `\exit` and `\echo` commands. These commands are available in the script and interactive modes. See the [echo](https://docs.hazelcast.com/clc/latest/echo) and [exit](https://docs.hazelcast.com/clc/latest/exit) command documentation.
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* Added `--wait` flag to `viridian` commands.

> * Added the ability to change the interactive mode prompt and banner. See: [def](https://docs.hazelcast.com/clc/latest/def).

Reviewer: ```suggestion
* Added the ability to change the interactive mode prompt and banner. See the [def](https://docs.hazelcast.com/clc/latest/def) topic.
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

== Improvements

> * `clc demo map-setmany` command shows a progress bar and it is faster for small values.

Reviewer: ```suggestion
* `clc demo map-setmany` command shows a progress bar and is faster for small values.
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89

* `\echo` command takes 0 or more arguments.

> * Viridian commands display an API key selector if the there exists more than one API keys.

Reviewer: ```suggestion
* Viridian commands display an API key selector if more than one API key exists.
```

Author: Fixed at: https://github.com/hazelcast/clc-docs/pull/13/commits/9febcbe07c44f071e16d557213df687261458d89
